infrastructure/src/main/java/org/corfudb/infrastructure/log/AbstractLogSegment.java

```diff
@@ -131,18 +139,15 @@
      * @param logData the LogData to append
      * @return metadata for the written record
      */
-    AddressMetaData writeRecord(long address, LogData logData,
-                                MultiReadWriteLock segmentLock) throws IOException {
+    AddressMetaData writeRecord(long address, LogData logData) throws IOException {
         LogEntry logEntry = getLogEntry(address, logData);
         Metadata metadata = getMetadata(logEntry);
 
         ByteBuffer record = getByteBuffer(metadata, logEntry);
         long channelOffset;
 
-        try (AutoCloseableLock ignored = segmentLock.acquireWriteLock(ordinal)) {
-            channelOffset = writeChannel.position() + METADATA_SIZE;
-            writeByteBuffer(writeChannel, record, logSizeQuota);
-        }
+        channelOffset = writeChannel.position() + METADATA_SIZE;
```

Comment:
Lock is removed because now we only compact addresses below committed tail, which means no one can write again to the segment. The caller are synchronized by batchProcessor in both master and GC.